File: challenge3.js
const gets = require('readline-sync').question;
const print = console.log;

const findSum = (num) => num.split('').reduce((acc, cur) => Number(acc) + Number(cur));

const q = Number(gets());

for (let t = 0; t < q; t++) {
  let [n, m] = gets().split(' ');
  let maxSum = findSum(m);
  for (let i = 1; i < m.length; i++) {
    if (m.length <= 1 || m.length <= i) break;
    m = (Number(m.slice(0, -i))-1).toString()+'9'.repeat(i);
    if (Number(m) < Number(n)) break;
    maxSum = Math.max(maxSum, findSum(m));
  }
  print(maxSum);
}

File: challenge3.py
import sys
cin = sys.stdin.readline

def cout(x):
  sys.stdout.write(str(x) + '\n')

def findSum(num):
  total = 0
  for char in num:
    total += int(char)
  return total


q = int(cin())

for t in range(q):
  n, m = cin().split()
  maxSum = findSum(m)
  for i in range(1, len(m)):
    if (len(m) <= 1 or len(m) <= i): #no values exist
      break

    m = str(int(m[:-i])-1)+'9'*i

    if (int(m) < int(n)):
      break
    maxSum = max(maxSum, findSum(m))
  cout(maxSum)